packages/compass-crud/src/components/option-write-selector.jsx

```diff
@@ -27,40 +26,9 @@ class OptionWriteSelector extends React.Component {
     isCollectionLevel: PropTypes.bool,
     onSelect: PropTypes.func.isRequired,
     options: PropTypes.object.isRequired,
-    tooltipId: PropTypes.string.isRequired
-  }
-
-  /**
-   * Subscribe to the state changing stores.
-   */
-  componentDidMount() {
-    this.unsubscribeWriteState = WriteStateStore.listen(this.writeStateChanged.bind(this));
-  }
-
-  /**
-   * Unsubscribe from the stores.
-   */
-  componentWillUnmount() {
-    this.unsubscribeWriteState();
-  }
-
-  /**
-   * Determine if the application is in a writable state.
-   *
-   * @returns {Boolean} If the application is writable.
-   */
-  isWritable() {
-    const isWritable = WriteStateStore.state.isWritable;
-    return isWritable;
-  }
-
-  /**
-   * Handle write state changes.
-   *
-   * @param {Object} state - The write state.
-   */
-  writeStateChanged(state) {
-    this.setState(state);
+    tooltipId: PropTypes.string.isRequired,
+    isWritable: PropTypes.bool.isRequired,
```

Comment:
See for now this component (which is only used for the Add Data button in crud that lets you pick insert/import) just gets isWritable as a prop rather than listening to a dedicated store.